src/sdk/interfaces/tier0/IMemAlloc.cpp

```diff
@@ -3,30 +3,203 @@
 #include <Include.h>
 #include <SDK/Interfaces/tier0/IMemAlloc.h>
 
-IMemAlloc* GetMemAlloc() {
-    static auto p_mem_alloc = *reinterpret_cast<IMemAlloc**>(GetProcAddress(GetModuleHandleA("tier0.dll"), "g_pMemAlloc"));
-    return p_mem_alloc;
+namespace {
+    IMemAlloc* MemAllocSystemInitialize() {
+        HMODULE tier0 = nullptr;
+
+        // Attempt to load tier0.dll
+        while (!tier0) {
+            tier0 = LoadLibraryA("tier0.dll");
+        }
+
+        IMemAlloc* g_pMemAlloc = nullptr;
+
+        // Continuously try to get the g_pMemAlloc pointer until it's successful
+        while (!g_pMemAlloc) {
+            g_pMemAlloc = *reinterpret_cast<IMemAlloc**>(GetProcAddress(tier0, "g_pMemAlloc"));
+
+            // If g_pMemAlloc is not found, try initializing it
+            if (!g_pMemAlloc) {
+                static const auto CMemAllocSystemInitialize = reinterpret_cast<void (*)()>(GetProcAddress(tier0, "CMemAllocSystemInitialize"));
+
+                if (CMemAllocSystemInitialize) {
+                    CMemAllocSystemInitialize();
+                }
+            }
+        }
+
+        return g_pMemAlloc;
+    }
+}; // namespace
+
+[[msvc::forceinline]] bool ValueIsPowerOfTwo(const std::size_t value) // don't clash with mathlib definition
```

Comment:
First of all, why would we implement it by ourselves
But if we still want to impl it, then please let the compiler optimize it by itself, no need to force inline

```suggestion
bool ValueIsPowerOfTwo(const std::size_t value) // don't clash with mathlib definition
```